crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -393,6 +393,16 @@ impl<'flox> Publish<'flox, NixAnalysis> {
             .map_err(PublishError::PathInfo)
             .await?;
 
+        let mut invalid_outputs = narinfos
+            .iter()
+            .filter(|n| !n.valid)
+            .map(|n| n.path.to_string_lossy().into_owned())
+            .peekable();
+
+        if invalid_outputs.peek().is_some() {
+            Err(PublishError::PathsNotCached(invalid_outputs.collect()))?;
```

Comment:
Failing in this case makes the method a bit more publish specific. That's fine for now, but I think the method comment should be updated

Reply:
I agree its publish specific, but probably fine for now since the struct is not visible outside of the publish module anyway.
If we use it more generically I agree that we pull that logic up.

Will update the comment